service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java

```diff
@@ -96,40 +105,47 @@ public void handlePaymentRedirectRequest(String externalId, String psuId,
       return;
     }
 
-    PisAuthorization payment = pisAuthorization.get();
+    PisAuthorization paymentAuth = pisAuthorization.get();
 
     Optional<List<PisPaymentData>> pisPaymentDataList = pisPaymentDataRepository
-        .findByPaymentId(payment.getPaymentData().getPaymentId());
+        .findByPaymentId(paymentAuth.getPaymentData().getPaymentId());
 
     if (!pisPaymentDataList.isPresent() && pisPaymentDataList.get().isEmpty()) {
```

Comment:
The two conditions must be connected by `||`, right?

Reply:
done